### PR TITLE
Expose Codex client session metadata

### PR DESCRIPTION
**Pull request by asubbarao**
## Problem

Agent Data could read only part of the locally stored Codex data. Codex CLI plans, todos, prompt history, and daily activity were unavailable, and Codex Work threads stored in Codex's local SQLite databases were not exposed. Grok tool-call rows could also share an identifier, causing downstream consumers to collapse distinct calls.

## Change

- Read Codex CLI conversations, plans, todos, history, and daily statistics.
- Read Codex Work conversations, plans, todos, history, and daily statistics from its durable local thread store.
- Expose Codex client originator, client source, and reasoning effort when present.
- Give fan-out Grok tool calls distinct identifiers.
- Update function documentation and schema validation.

Codex Remote and Chat sources are recognized when durable local thread records exist; this machine has no such records yet.

## Validation

- `cargo test` (7 tests)
- Debug SQLLogic suite (24 tests)
- Release SQLLogic suite (24 tests)
- `cargo clippy` introduces no warnings beyond the 11 already present on `main`

### CODE REVIEW

File: src/conversations.rs [hunk 1]
```diff
@@ -1055,14 +1073,15 @@ impl Conversations {
                                 row.uuid.take(),
                                 session_uuid,
                                 file_line,
+                                row.tool_use_id.as_deref(),
```
Comment (asubbarao) [inline src/conversations.rs]: **[suggestion]** `grok_row_uuid` is always called with `row.tool_use_id.as_deref()`, so any row that carries `tool_use_id` — including one-per-line `tool_result` rows — switches from `{session}:{line}` to `{session}:{line}:tool:{id}`. The README and function comment only describe this form for fan-out `tool_call` rows; `tool_result` UUID churn is an undocumented side effect of the fan-out fix.

**Suggestion:** Pass `tool_use_id` only when `row.message_type == "tool_call"` (preserving prior `tool_result` synthetic IDs), or explicitly document that `tool_result` UUIDs also use the `:tool:` suffix.

File: src/conversations.rs [hunk 2]
```diff
@@ -935,9 +943,19 @@ impl Conversations {
...
+    /// Prefer a real message id. A fan-out tool call needs its own identity,
+    /// otherwise every tool emitted from one chat_history line aliases the
+    /// same synthetic UUID and downstream consumers silently lose rows.
+    fn grok_row_uuid(
+        existing: Option<String>,
+        session_id: &str,
+        line_number: i64,
+        tool_use_id: Option<&str>,
+    ) -> String {
+        existing.unwrap_or_else(|| match tool_use_id {
```
Comment (asubbarao) [inline src/conversations.rs]: **[suggestion]** If multiple `tool_calls` on one assistant line omit `id` (`GrokToolCall.id` is `Option<String>`), `tool_use_id` is `None` and every fan-out row still synthesizes the shared `{session_id}:{line_number}` UUID — the collision this PR aims to fix remains for that edge case.

**Suggestion:** When `tool_use_id` is missing, fall back to a per-emission discriminator (for example tool index) so fan-out rows stay unique even without ids.

File: test/sql/column_validation.test [hunk 3]
```diff
@@ -15,6 +15,20 @@ WHERE typeof(source) != 'VARCHAR';
...
+# Codex client metadata columns are VARCHAR when populated.
+query I
+SELECT COUNT(*) FROM read_conversations(path='test/data_codex')
+WHERE typeof(client_originator) != 'VARCHAR' OR typeof(client_source) != 'VARCHAR';
+----
+0
+
+# Other providers leave Codex-specific client metadata NULL.
+query I
+SELECT COUNT(*) FROM read_conversations(path='test/data_claude')
```
Comment (asubbarao) [inline test/sql/column_validation.test]: **[suggestion]** The “other providers leave client_* NULL” check only scans `test/data_claude`. Copilot/Cursor/Gemini/Grok paths are not asserted, so a future accidental backfill on another loader would not be caught by this suite.

**Suggestion:** Extend the NULL invariant to the other fixture roots (or one `UNION ALL` across them) so the Codex-only contract is enforced for every provider under test.

File: README.md [hunk 4]
```diff
@@ -230,9 +232,10 @@ Reads conversation/event data.
 > Sessions without `updates.jsonl` (or without `usage`) keep token columns NULL.
 > `cache_creation_tokens` is never set for Grok.
 >
-> **Synthetic uuid form:** `{session_id}:{line_number}` (1-based chat_history line).
-> Multi-row fan-out from one assistant line (text + tool_call rows) shares that
-> line's uuid. Prefer real `reasoning.id` when present.
+> **Synthetic uuid form:** ordinary rows use `{session_id}:{line_number}` (1-based
+> `chat_history` line). A fan-out `tool_call` uses
+> `{session_id}:{line_number}:tool:{tool_use_id}`; the optional assistant text row
```
Comment (asubbarao) [inline README.md]: **[nit]** Synthetic UUID docs say only fan-out `tool_call` rows use `{session_id}:{line_number}:tool:{tool_use_id}`. That understates current runtime behavior for `tool_result` and can mislead consumers re-keying on `uuid`.

**Suggestion:** Align the README sentence with whatever policy is chosen for the `tool_result` UUID comment on `conversations.rs`.

File: src/conversations.rs (hunk 2 above)
Comment (asubbarao) [inline src/conversations.rs]: **[suggestion]** `grok_row_uuid` now prefers any `existing` id and only synthesizes `:tool:{tool_use_id}` in the `None` branch. Call sites today are safe: only `GrokMessage::Reasoning` sets `uuid`, and assistant fan-out tool rows leave `uuid` unset so they get distinct synthetic ids. However this is still a breaking change versus the prior “one uuid per chat_history line” contract (including `tool_result` rows, which also carry `tool_use_id` and therefore also pick up the `:tool:` suffix). The PR title/body only describe the Codex columns; downstream FTS/consumers that keyed on shared line uuids need an explicit migration note.

**Suggestion:** Call out the Grok UUID uniqueness break in the PR description (and any release notes). Optionally narrow the `:tool:` suffix to `message_type == "tool_call"` if `tool_result` should keep `{session}:{line}`, and/or update the README synthetic-uuid blurb so it matches whichever rule ships.

File: test/sql/codex_conversations.test [hunk 5]
```diff
@@ -138,6 +138,19 @@ SELECT DISTINCT git_branch, cwd, version, repository FROM read_conversations(pat
...
+# Raw Codex client metadata stays separate from normalized source='codex'.
+query II
+SELECT DISTINCT client_originator, client_source
+FROM read_conversations(path='test/data_codex');
+----
+Codex CLI	vscode
+
+query I
+SELECT COUNT(*) FROM read_conversations(path='test/data_codex')
+WHERE source != 'codex' OR client_originator != 'Codex CLI' OR client_source != 'vscode';
```
Comment (asubbarao) [inline test/sql/codex_conversations.test]: **[suggestion]** The invariant query uses `client_originator != 'Codex CLI' OR client_source != 'vscode'`. Under SQL three-valued logic, a NULL client field makes the inequality NULL, so a partially-populated session can still yield `COUNT(*) = 0` and look green. The preceding `SELECT DISTINCT client_originator, client_source` expectation does catch all-NULL / mixed values, so this is coverage weakness rather than a current false pass on the fixture.

**Suggestion:** Prefer `IS DISTINCT FROM` (or `WHERE NOT (source = 'codex' AND client_originator = 'Codex CLI' AND client_source = 'vscode')`) so NULL client metadata fails the count check on its own.

File: src/conversations.rs [hunk 6]
```diff
@@ -1189,6 +1208,7 @@ impl TableFunc for Conversations {
             vtab::varchar("git_branch"),    vtab::varchar("cwd"),
             vtab::varchar("version"),       vtab::varchar("stop_reason"),
             vtab::varchar("reasoning_effort"), vtab::varchar("repository"),
+            vtab::varchar("client_originator"), vtab::varchar("client_source"),
```
Comment (asubbarao) [inline src/conversations.rs]: **[nit]** Adding `client_originator` / `client_source` grows `read_conversations` from 29 to 31 columns, but `test/sql/edge_cases.test` still comments “Conversations: 29 columns …” under a “Column count validation” heading. That query only `COUNT(*)`s a `LIMIT 0` subquery (always 0 rows) and never asserts schema width, so CI stays green while the comment drifts.

**Suggestion:** Update the comment to 31 (or replace the no-op with a real width check such as `SELECT COUNT(*) FROM (DESCRIBE SELECT * FROM read_conversations(...))`).

File: README.md [hunk 7]
```diff
@@ -230,9 +232,10 @@ Reads conversation/event data.
 > Sessions without `updates.jsonl` (or without `usage`) keep token columns NULL.
 > `cache_creation_tokens` is never set for Grok.
 >
-> **Synthetic uuid form:** `{session_id}:{line_number}` (1-based chat_history line).
-> Multi-row fan-out from one assistant line (text + tool_call rows) shares that
-> line's uuid. Prefer real `reasoning.id` when present.
+> **Synthetic uuid form:** ordinary rows use `{session_id}:{line_number}` (1-based
+> `chat_history` line). A fan-out `tool_call` uses
```
Comment (asubbarao) [inline README.md]: **[nit]** Docs say a fan-out `tool_call` uses `{session}:{line}:tool:{tool_use_id}` while the assistant text row keeps the base id. Implementation keys only on `tool_use_id.is_some()`, so `tool_result` rows (which set `tool_use_id` but are not fan-out) also get the `:tool:` form. Mild docs/code mismatch; behavior is still unique per line.

**Suggestion:** Either document that any row with `tool_use_id` gets the suffix, or gate synthesis on `message_type == "tool_call"` to match the README.

**Comment by asubbarao** [comment]
Closing as too large to review as one change (1,580 lines, 26 files, spanning Codex CLI plans/todos/history/stats, the Codex Work SQLite store, client metadata and a Grok tool-call id fix). The branch is kept; these are separable and each piece is reviewable on its own.
